dracut/image-boot/eos-image-boot-generator

```diff
@@ -0,0 +1,86 @@
+#!/bin/sh
+# Copyright (C) 2016 Endless Mobile, Inc.
+# Licensed under the GPLv2
+#
+# Support booting from an image file hosted on a filesystem.
+# We parse the kernel parameters requesting this functionality and configure
+# other systemd units to go about setting up the block devices and mounting
+# the target filesystem.
+
+type getarg >/dev/null 2>&1 || . /lib/dracut-lib.sh
+
+GENERATOR_DIR="$1"
+
+image_device=$(getarg endless.image.device=)
+image_path=$(getarg endless.image.path=)
+
+[ -z "${image_device}" -o -z "${image_path}" ] && exit 0
+
+case "${image_device}" in
+PARTUUID=*)
+  image_device=/dev/disk/by-partuuid/${image_device#PARTUUID=}
```

Comment:
PARTUUID= and UUID= is the standard way of passing this info from the bootloader, so I think it makes sense to follow that.